subsys/bluetooth/mesh/health_cli.c

```diff
@@ -486,9 +478,15 @@ int bt_mesh_health_cli_fault_clear(struct bt_mesh_health_cli *cli, struct bt_mes
 	bt_mesh_model_msg_init(&msg, OP_HEALTH_FAULT_CLEAR);
 	net_buf_simple_add_le16(&msg, cid);
 
-	return model_ackd_send(cli->model, ctx, &msg,
-			       (!test_id && (!faults || !fault_count)) ? NULL : &cli->ack_ctx,
-			       OP_HEALTH_FAULT_STATUS, &param);
+	struct bt_mesh_msg_rsp_ctx rsp = {
+		.ack = &cli->ack_ctx,
+		.op = OP_HEALTH_FAULT_STATUS,
+		.user_data = &param,
+		.timeout = msg_timeout,
+	};
+
+	return bt_mesh_msg_ackd_send(cli->model, ctx, &msg,
+				     (!test_id && (!faults || !fault_count)) ? NULL : &rsp);
```

Comment:
```suggestion
	#define BT_MESH_MSG_RSP(cond, ack, op, user_data, timeout) \
				(cond) ? NULL : \
				(struct bt_mesh_msg_rsp_ctx ){ ack, op, user_data, timeout }

	return bt_mesh_msg_ackd_send(cli->model, ctx, &msg,
				     BT_MESH_MSG_RSP((!test_id && (!faults || !fault_count)),
		     		     		     &cli->ack_ctx,
		     		     		     OP_HEALTH_FAULT_STATUS,
		     		     		     &param,
		     		     		     msg_timeout));
```

Reply:
I'd like this way to look like code more brief, and not increase code size.

Reply:
As far as I understand this won't change code size, but just the form of the code. And this I personally don't like.

Reply:
True, not change code size, but will to code more brief, at least i think.

Reply:
IMO this makes the code more difficult to read as you may think that response is always passed, but it is not.